### PR TITLE
feat(emails): needs improvements emails

### LINKED ISSUE #2668: [feature request] add moderation feedback display for map pins that 'need improvement' or are 'rejected'
**Is your feature request related to a problem? Please describe.**
Map pins that have been 'rejected' / need updates before they can be approved need to be able to show any moderation feedback left by admins so that the user can make changes and re-submit.

**Describe the solution you'd like**
The feedback comments should be shown in 2 places:
- On the map pin card displayed on the map along with a link to edit their map pin
- On the profile settings page where the map pin is edited it should display the status and if 'rejected' the feedback comments.

Describe alternatives you've considered
N/A

### PR DESCRIPTION
**Pull request by evakill**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

adds needs improvements transactional emails with moderator comments

will wait to merge until #2820 

## Git Issues

Closes #2668

## Screenshots/Videos

Screen Shot 2023-09-30 at 9 19 42 AM
Screen Shot 2023-09-30 at 9 14 13 AM

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: functions/src/emailNotifications/templates/index.ts [hunk 1]
```diff
@@ -8,6 +8,8 @@ type SupportedEmailTemplates =
   | 'how-to-rejected'
   | 'map-pin-rejected'
   | 'verified-badge-added'
+  | 'how-to-needs-improvements'
+  | 'map-pin-needs-improvements'
```
Comment (thisislawatts) [inline functions/src/emailNotifications/templates/index.ts:12]: **thought:** I wonder if we can auto generate this type based on the files within this directory 🤔

Comment (evakill) [reply functions/src/emailNotifications/templates/index.ts:12]:
```
export function getEmailHtml(emailType: string, ctx: {}) {
  fs.readdir(__dirname, (err, files) => {
    if (err) console.log(err)
    else {
      if (!files.includes(`${emailType}.html`)) {
        throw new Error(`Email template ${emailType}.html does not exist.`)
      }
    }
  })
 ...
```
is this better?

Comment (evakill) [reply functions/src/emailNotifications/templates/index.ts:12]: ooh they have a synchronous method. might be able to generate the type after all

Comment (thisislawatts) [reply functions/src/emailNotifications/templates/index.ts:12]: I like the suggested template name that the type gives us at the moment. I might be overcomplicating things. Although the error handler you've added there should definitely be added! Much more robust implementation.

Comment (evakill) [reply functions/src/emailNotifications/templates/index.ts:12]: > I like the suggested template name that the type gives us at the moment

yeah, i think we'll only get that w the static definition. will keep as is and add the error handling

**Review by thisislawatts — APPROVED** [review]

**Pull request merged**